daffodil-io/src/main/scala/org/apache/daffodil/io/DirectOrBufferedDataOutputStream.scala

```diff
@@ -765,15 +818,30 @@ class DirectOrBufferedDataOutputStream private[io] (var splitFrom: DirectOrBuffe
       // no fragment bytes anywhere - just take the bytes
 
       val nBytes = (bufferNBits / 8).toInt
-      val nBytesPut = directDOS.putBytes(ba, 0, nBytes, finfo)
+      val nBytesPut = {
+        if (this.bufferingJOS.isFile)
+          directDOS.putFile(
+            this.bufferingJOS.tempFile.toPath,
+            this.bufferingJOS.bytesWritten * 8, 4096, finfo)
+        else
+          directDOS.putBytes(ba, 0, nBytes, finfo)
```

Comment:
Where does ba come from? Seems like we should only be gettingg that if this is not a file based output stream?